umapi_client/functional.py

```diff
@@ -329,39 +317,32 @@ def maybe_split_groups(self, max_groups):
         new_commands = []
         # return True if we split at least once
         maybe_split = False
+        valid_step_keys = ['add', 'addRoles', 'remove']
         for command in self.commands:
+            # commands are assumed to contain a single key
             step_key, step_args = next(six.iteritems(command))
-            if step_key not in [StepKeys.add.name, StepKeys.remove.name, StepKeys.addRoles.name]:
+            if step_key not in valid_step_keys:
+                new_commands.append(command)
                 continue
-            split_commands = self._split_groups(step_key, command, max_groups)
-            if len(split_commands) > 1:
-                maybe_split = True
+            split_commands = [command]
+            while True:
+                new_command = {step_key: {}}
+                if not isinstance(command[step_key], dict):
```

Comment:
combine this into the test on line 34:
```python
if step_key not in valid_step_keys or not isinstance(step_args, dict):
```
That way the logic is cleaner in the loop and you are taking advantage of having the `step_args`.